File: src/main/java/com/practica/crean/arquitecture/infraestructure/persistence/mapper/EstudianteMapper.java
package com.practica.crean.arquitecture.infraestructure.persistence.mapper;

import com.practica.crean.arquitecture.core.domain.Estudiante;
import com.practica.crean.arquitecture.infraestructure.persistence.entity.EstudianteEntity;
import org.mapstruct.Mapper;

import java.util.List;

/*
 * MapStruct es una biblioteca de Java que genera automáticamente
 * mapeadores para convertir entre diferentes tipos de objetos,
 * utilizando interfaces anotadas para definir las reglas de mapeo.
 * */
@Mapper(componentModel = "spring")
public interface EstudianteMapper {

    Estudiante toEstudiante(EstudianteEntity entity);

    EstudianteEntity toEstudianteEntity(Estudiante estudiante);

    List<Estudiante> toListaEstudiantes(List<EstudianteEntity> entity);

    List<EstudianteEntity> toListaEstudiantesEntity(List<Estudiante> entity);

}

File: src/main/kotlin/com/practica/crean/arquitecture/infraestructure/persistence/mapper/EstudianteMapper.kt
package com.example.demo.infraestructure.persistence.mapper

import com.example.demo.core.domain.Estudiante
import com.example.demo.infraestructure.persistence.entity.EstudianteEntity
import org.mapstruct.Mapper

/*
* MapStruct es una biblioteca de Java que genera automáticamente
* mapeadores para convertir entre diferentes tipos de objetos,
* utilizando interfaces anotadas para definir las reglas de mapeo.
* */
@Mapper(componentModel = "spring")
interface EstudianteMapper {

    fun toEstudiante(estudianteEntity: EstudianteEntity): Estudiante

    fun toEstudianteEntity(estudiante: Estudiante): EstudianteEntity

    fun toListaEstudiante(estudiantes: List<EstudianteEntity>): List<Estudiante>

    fun toListaEstudianteEntity(estudiantes: List<Estudiante>): List<EstudianteEntity>

}